sklearnex/linear_model/incremental_ridge.py

```diff
@@ -0,0 +1,418 @@
+        is_underdetermined = self.n_samples_seen_ < self.n_features_in_ + int(
+            self.fit_intercept
+        )
+        if is_underdetermined:
+            raise ValueError("Not enough samples to finalize")
+        self._onedal_estimator.finalize_fit()
+        self._save_attributes()
+        self._need_to_finalize = False
+
+    def _onedal_fit(self, X, y, queue=None):
+        if sklearn_check_version("1.2"):
+            self._validate_params()
+
+        if sklearn_check_version("1.0"):
+            X, y = self._validate_data(
+                X,
+                y,
+                dtype=[np.float64, np.float32],
+                copy=self.copy_X,
+                multi_output=True,
+                ensure_2d=True,
+            )
+        else:
+            check_X_y(X, y, multi_output=True, y_numeric=True)
+
+        n_samples, n_features = X.shape
+
+        is_underdetermined = n_samples < n_features + int(self.fit_intercept)
+        if is_underdetermined:
+            raise ValueError("Not enough samples to run oneDAL backend")
+
+        if self.batch_size is None:
+            self.batch_size_ = 5 * n_features
+        else:
+            self.batch_size_ = self.batch_size
+
+        self.n_samples_seen_ = 0
+        if hasattr(self, "_onedal_estimator"):
+            self._onedal_estimator._reset()
+
+        for batch in gen_batches(n_samples, self.batch_size_):
+            X_batch, y_batch = X[batch], y[batch]
+            self._onedal_partial_fit(X_batch, y_batch, check_input=False, queue=queue)
+
+        if sklearn_check_version("1.2"):
+            self._validate_params()
+
+        # finite check occurs on onedal side
+        self.n_features_in_ = n_features
+
+        if n_samples == 1:
+            warnings.warn(
+                "Only one sample available. You may want to reshape your data array"
+            )
+
+        self._onedal_finalize_fit()
+
+        return self
+
+    def partial_fit(self, X, y, check_input=True):
+        """
+        Incrementally fits the linear model with X and y. All of X and y is
+        processed as a single batch.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training data, where `n_samples` is the number of samples and
+            `n_features` is the number of features.
+
+        y : array-like of shape (n_samples,) or (n_samples, n_targets)
+            Target values, where `n_samples` is the number of samples and
+            `n_targets` is the number of targets.
+
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+
+        dispatch(
+            self,
+            "partial_fit",
+            {
+                "onedal": self.__class__._onedal_partial_fit,
+                "sklearn": None,
+            },
+            X,
+            y,
+            check_input=check_input,
+        )
+        return self
+
+    def fit(self, X, y):
```

Comment:
Maybe I'm opening pandora's box here but Ridge takes sample_weight https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/linear_model/_ridge.py#L911 . This is also the case for preview Ridge.  Please add a keyword for conformance purposes and add an error if its not None. I know that may seem unnecessary, but generally trying to conform to related estimators is top priority.  You can let me know if you find that unneccessary.